### ISSUE TITLE
bug: < sign and substring after that get filtered from revision field "changes"

### ISSUE THREAD

**Issue by Knorrke** [serlo-org/athene2#728]
When someone uses a `<` symbol in the "changes" field of a revision, then this and everything after it will be filtered and isn't visible e.g. for reviewers or in the history. It's probably already filtered when saving, not when displaying it.

I see two options: 
1. We could maybe improve that filter to distinguish between html (which we want removed because of script-injection and breaking format and stuff) and use of `<` `>` signs.
2. Escape the strings instead of filtering? E.g. replace `<` and `>` by `&lt;` and `&gt;` when saving. Not sure if this introduces weaknesses, @inyono what do you think?

**Comment by inyono** [comment]
I prefer escaping over stripping (and we'd have to cover each exception differently). Escaping should be fine for this use case, especially, since there's a good reason to use a couple of special chars in math context.

**Comment by inyono** [comment]
To approach this:
* The forms are described in `src/module/Entity/src/Entity/Form`
* In each form, there are a couple of input filters defined (e.g. is a field required, and do we want to process the user entered content in any way)
* Change the input filter for text-y inputs (e.g. `changes`, maybe others, too) from `StripTags` to something more sensible. You can find information about the input filters in the Zend 2 documentation, especially https://framework.zend.com/manual/2.4/en/modules/zend.input-filter.intro.html and following sections.
* Test your changes by editing an entity (e.g. article) and checking if changes to the relevant fields are handled correctly

**Comment by Simon128** [comment]
Possible Solution (explicitly for the revision field 'changes' e.g. in [ArticleForm.php](https://github.com/serlo-org/athene2/blob/master/src/module/Entity/src/Entity/Form/ArticleForm.php)):
```php
//old: $inputFilter->add(['name' => 'changes', 'required' => false, 'filters' => [['name' => 'StripTags']]]);
//new:
$inputFilter->add(['name' => 'changes', 'required' => true, 'filters' => [
            ['name' => 'PregReplace', 'options' => ['pattern' => '/</','replacement' => '&lt']],
            ['name' => 'PregReplace', 'options' => ['pattern' => '/>/','replacement' => '&gt']],
]]);
```

For which other inputs should the 'StripTags' filter be replaced? @inyono you wrote something about maybe others, too?

**Comment by inyono** [comment]
Hey @Simon128. Thanks for looking into that.

Regarding your proposed change: I was hoping that Zend already has a filter to html-encode any special chars, not only `<` and `>`. Since we have `changes` in multiple places, we at least need to create our own filter for that (so that we only need to change it in one place) if Zend doesn't have a suitable one, itself.

Regarding other inputs: I was thinking about "Didaktische Überlegungen" (i.e. `reasoning`) but probably most other fields using `StripTags` would be better of with the new filter, too. Feel free to check the other fields and list fields where it does / does not make sense in your opinion. (We'll review that later ofc :))

**Comment by inyono** [comment]
Closed by #729
